Library/Homebrew/formulary.rb

```diff
@@ -222,7 +223,21 @@ def self.loader_for(ref)
       return FromPathLoader.new(ref)
     end
 
-    formula_with_that_name = Formula.path(ref)
+    if ref.start_with?(HOMEBREW_CELLAR) && File.directory?(ref)
+      rack = Pathname.new(ref)
+      name = rack.basename.to_s
+      kegs = rack.subdirs.map { |d| Keg.new(d) }
+      raise FormulaUnavailableError.new(name) if kegs.empty?
+      keg = kegs.detect(&:linked?) || kegs.detect(&:optlinked?) || kegs.max_by(&:version)
+      tap = Tab.for_keg(keg).tap
+      if tap == "Homebrew/homebrew"
+        return FormulaLoader.new(name, Formulary.core_path(name))
+      else
+        return TapLoader.new("#{tap.sub "homebrew-", ""}/#{name}")
+      end
+    end
```

Comment:
Yeh, this seems pretty great to me. Finally my `brew outdated` is useful again because it's not trying to upgrade versions of my stuff installed from taps (that are intentionally old versions).